packages/rest/test/integration/http-handler.integration.ts

```diff
@@ -324,14 +347,45 @@ describe('HttpHandler', () => {
       const body = {key: givenLargeRequest()};
       rootContext
         .bind(RestBindings.REQUEST_BODY_PARSER_OPTIONS)
-        .to({limit: 4 * 1024 * 1024}); // Set limit to 4MB
+        .to({limit: '4mb'}); // Set limit to 4MB
       return client
         .post('/show-body')
         .set('content-type', 'application/json')
         .send(body)
         .expect(200, body);
     });
 
+    it('allows request body parser extensions', () => {
+      const body = '<key>value</key>';
+
+      /**
+       * A mock-up xml parser
+       */
+      class XmlBodyParser implements BodyParser {
+        name: string = 'xml';
+        supports(mediaType: string) {
+          return !!is(mediaType, 'xml');
+        }
+
+        async parse(request: Request) {
+          return {value: {key: 'value'}};
+        }
+      }
+
+      // Register a request body parser for xml
```

Comment:
(1) Please add a sugar API for registering body parsers. 

Typical usage in a REST application:

```ts
class TodoApplication extends /*mixins*/ RestApplication {
  constructor() {
    // ...
  
    this.bodyParser(XmlBodyParser);
  }
}
```

(2) Add a new test (or extend this one) to show the idiomatic solution for passing configuration options (e.g. max body size) to custom request parsers.